app/templates/results.html

```diff
@@ -48,6 +48,14 @@
                 ctx.fillText(event.target.value, 10, 50);
             });
         });
+        function downloadImage(canvasId) {
```

Comment:
Will this pick up the image and the text that I want to overlay?

Reply:
Yes, the `downloadImage` function will indeed capture both the image and the overlayed text. Here's how it works:

1. **Canvas Setup**: Each image is displayed within a `<canvas>` element. The text input by the user is dynamically drawn onto this canvas using the `fillText` method of the canvas's 2D rendering context. This happens in the event listener for the text input's `input` event.

2. **Downloading the Canvas Content**: When the `downloadImage` function is called, it accesses the canvas by its ID. It then uses the `toDataURL` method to convert the entire canvas content (which includes both the image and the overlayed text) into a data URL representing the image in PNG format.

3. **Triggering the Download**: The function creates an `<a>` element with the `href` attribute set to this data URL and triggers a click event on this link. This causes the browser to download the content of the canvas as an image file.

This approach ensures that whatever is currently rendered on the canvas (both the image and any text drawn on it) will be included in the downloaded file.